File: cloud_testing/cloud_api_poc/src/main/java/cloud_api_poc/GcAndroidMatrix.java
package cloud_api_poc;

import com.google.common.collect.Lists;
import com.google.testing.model.AndroidMatrix;

public abstract class GcAndroidMatrix {

  private GcAndroidMatrix() {}

  public static AndroidMatrix build(
      String modelIds, String versionIds, String locales, String orientations) {
    AndroidMatrix androidMatrix = new AndroidMatrix();
    androidMatrix.setAndroidModelIds(Lists.newArrayList(modelIds));
    androidMatrix.setAndroidVersionIds(Lists.newArrayList(versionIds));
    androidMatrix.setLocales(Lists.newArrayList(locales));
    androidMatrix.setOrientations(Lists.newArrayList(orientations));
    return androidMatrix;
  }
}

File: cloud_testing/kotlin_poc/src/main/kotlin/GcAndroidMatrix.kt
import com.google.common.collect.Lists
import com.google.testing.model.AndroidMatrix

object GcAndroidMatrix {

    fun build(
            modelIds: String, versionIds: String, locales: String, orientations: String): AndroidMatrix {
        val androidMatrix = AndroidMatrix()
        androidMatrix.androidModelIds = Lists.newArrayList(modelIds)
        androidMatrix.androidVersionIds = Lists.newArrayList(versionIds)
        androidMatrix.locales = Lists.newArrayList(locales)
        androidMatrix.orientations = Lists.newArrayList(orientations)
        return androidMatrix
    }
}
